### ISSUE TITLE
Port meteor cookbook articles from hackpad?

### ISSUE THREAD

**Issue by nate-strauser** [meteor/guide#13]
I wrote these articles for MDG some time ago.

https://meteor.hackpad.com/Meteor-Cookbook-Using-Dates-and-Times-qSQCGFc06gH
https://meteor.hackpad.com/Meteor-Cookbook-Reactive-D3-Visualizations-YUR9JT4mrm9
https://meteor.hackpad.com/Meteor-Cookbook-Filepicker.io-Uploads-and-Image-Conversion-hIpSwJQV3HJ

A couple more are less polished / still drafts here https://github.com/nate-strauser/meteor-cookbook

Do you think this type of content belongs in the guide?  I'd be happy to help update these and get them ported over.

**Comment by stubailo** [comment]
Those are really great articles! I'd love to find a place for them somewhere in the guide. We should think about how they would fit into a developer's workflow. For example, maybe the D3 visualization thing is part of UX? File uploads is pretty clearly something people need. I don't know if FIlepicker is the most versatile solution we could recommend, but we should take a look (funny story, Filepicker was started by my college friend!). Dates and times is great content, but I think it should be split between multiple articles - I'd rather organize content around something else other than data types.

In short I don't have a great answer yet, let's revisit once we have a bit more structure and content in place.

**Comment by nate-strauser** [comment]
ok.  i'm in no rush, just wanting to help this project along.

i'm not wed to filepicker.  i typically use uploadcare now since it's more
affordable for small sites.  article could easily be rewritten to be more
'cloud based file upload services' and feature both filepicker and
uploadcare.  could have some nice related articles covering packages that
do uploads (collectionfs, slingshot, etc) and direct uploads (s3, dropbox?).

also, i've got a list of topics in the same vein as these articles.  might
be useful to stub about articles we think would be worth it.  then anyone
can just jump in and get them going.

**Comment by stubailo** [comment]
Have you seen the outlines? https://github.com/meteor/guide/blob/gh-pages/outlines.md

If there is something else you think we should have as a top-level article, please submit a PR or just let me know.

I'd like to build at least some of those articles first, and then maybe add a section for random cookbook articles if it makes sense.

**Comment by tmeasday** [comment]
I'm inclined to say that more cookbooky articles should maybe live on their own site. I'd be conscious of minimising the maintenance burden of this guide effort. Let's be realistic about it.

I don't think people need to see cookbook articles endorsed by MDG?

(I have no idea of the history of these articles btw. I'd not seen them before, which is a bit surprising!)

**Comment by tmeasday** [comment]
Definitely something to add to the puzzle though. Kind of like the Meteor Manual, I feel like this is great content but it's outside of the scope of the guide...

**Comment by stubailo** [comment]
I definitely think file uploads should be in the guide. A huge variety of apps need some way to upload files, and I think there are some good working solutions out there we could document.

**Comment by awatson1978** [comment]
Those articles are classics.  :)

Here are some additional resources to add to those articles, including the definitive Mongo presentation on time-series-data; and a demo app that implements a time-series data graph using D3 and date-time elements.

https://www.mongodb.com/presentations/mongodb-time-series-data
https://github.com/awatson1978/rest-analytics-pipeline
http://rest-analytics-pipeline.meteor.com/


There's also the following article, which documents 4 additional file upload patterns.
https://github.com/awatson1978/meteor-cookbook/blob/master/cookbook/file-uploads.md

**Comment by tmeasday** [comment]
https://forums.meteor.com/t/community-contributed-docs-examples-from-stackoverflow/20888/5
